src/util/expr.cpp

```diff
@@ -539,3 +539,134 @@ const_unique_depth_iteratort exprt::unique_depth_cbegin() const
 { return const_unique_depth_iteratort(*this); }
 const_unique_depth_iteratort exprt::unique_depth_cend() const
 { return const_unique_depth_iteratort(); }
+
+static bool precedence(const exprt &a, const exprt &b)
+{
+  // we use the precendences that most readers expect
+  // (i.e., the ones you learn in primary school),
+  // and stay clear of the surprising ones that C has
```

Comment:
This comment would make a perfect doxygen comment, if only you moved it two lines up (and added a third slash).